cirq/ops/linear_combinations.py

```diff
@@ -315,6 +316,90 @@ def copy(self) -> 'PauliSum':
         factory = type(self)
         return factory(self._linear_dict.copy())
 
+    def expectation_from_wavefunction(
+            self,
+            state: np.ndarray,
+            qubit_map: Dict[raw_types.Qid, int]) -> float:
+        """Evaluate the expectation of this PauliSum given a wavefunction.
+
+        See `PauliString.expectation_from_wavefunction`.
+        """
+        if any([abs(p.coefficient.imag) > 0.0001 for p in self]):
+            raise NotImplementedError(
+                "Cannot compute expectation value of a non-Hermitian "
```

Comment:
Is this actually the case? I'd expect the expectation to be just be scaled by the complex value.

It also seems like this would be handled fine from within PaulSitring.

@dabacon opinion on this? Is the concept of the expectation well defined when there's imaginary terms? Is it better to have guard rails?

Reply:
Yea I agree that non-Hermitian operators should work fine if allowed, so the error message isn't quite correct.

My main arguments for disallowing non-Hermitian operators is that the usual definition for variance of a Hermitian operator B (Var[B] = <B**2> - <B>**2) breaks down. For example, the operator B=iX has Var[B] = i for the state |0> using that standard formula.

The accepted answer at his [link ](https://physics.stackexchange.com/questions/82613/why-are-only-real-things-measurable) shows that non-Hermitian operator expectations don't add any outcomes that couldn't have been accomplished with Hermitain operators, though that doesn't mean they couldn't be convenient for some purpose.

Reply:
By the way, it can have imaginary coefficients and still be Hermitian (if the Hermitian-conjugated term is also in there). I lean toward handling the non-Hermitian case but also want to see if @dabacon has an opinion.

Reply:
Making it just Hermitian make sense when thinking of this as "expectation value of an observable in quantum theory".  For non-Hermitian operators, sometimes you can think of them as "observables" with complex eignevalues.  But sometimes you end up with non-Hermitian operators that don't a complete set of eigenvectors than span the space, so then it is really hard to interpret this as an observable. 

That said, it definitely makes sense to define <x|A|x> for a non-hermitian operator.  My suggestion would be to allow this via an explicit parameter.  i.e. normally it requires Hermitian and throws an error that says if you want to calculate for a non-hermitian operator, set the parameter appropriate.